File: kotlin/techverito/src/main/domain/Product.java
package domain;

public class Product {
    private Type type;
    private int price;

    public Product(Type type, int price) {
        this.type = type;
        this.price = price;
    }

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "domain.Product{" +
                "type=" + type +
                ", price=" + price +
                '}';
    }
}

File: kotlin/product-publicist-sapient/src/main/kotlin/com/myproject/domain/Product.kt
package com.myproject.domain

import java.math.BigDecimal

open class Product(val id: Int, var name: String, var price: BigDecimal, var color: String)

class Shirt(id: Int, name: String, price: BigDecimal, color: String) : Product(id, name, price, color) {
    override fun toString(): String {
        return "Shirt(id=$id, name=$name, price=$price, color=$color)"
    }
}

data class Person(val id: Int, val name: String)
